cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,392 @@
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail/types"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/osdctl/pkg/osdCloud"
+	"github.com/openshift/osdctl/pkg/osdctlConfig"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// Configuration struct for parsing configuration options
+type Configuration struct {
+	Ignore struct {
+		Users []string `mapstructure:"users_with"`
+	} `mapstructure:"ignore"`
+}
+
+// LookupEventsoptions struct for holding options for event lookup
+type LookupEventsoptions struct {
+	clusterID string
+	since     string
+	pages     int
+	cluster   *cmv1.Cluster
+	url       bool
+	raw       bool
+	all       bool
+}
+
+// RawEventDetails struct for holding raw event details
+type RawEventDetails struct {
+	EventVersion string `json:"eventVersion"`
+
+	UserIdentity struct {
+		AccountId      string `json:"accountId"`
+		Type           string `json:"type"`
+		SessionContext struct {
+			SessionIssuer struct {
+				Type     string `json:"type"`
+				UserName string `json:"userName"`
+				Arn      string `json:"arn"`
+			} `json:"sessionIssuer"`
+		} `json:"sessionContext"`
+	} `json:"userIdentity"`
+	EventRegion string `json:"awsRegion"`
+	EventId     string `json:"eventID"`
+}
+
+func newwrite_eventsCmd() *cobra.Command {
+	ops := newWrite_eventsOptions()
+	listEventsCmd := &cobra.Command{
+		Use:   "write-events",
+		Short: "Prints out all cloudtrail write events to console",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("")
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+
+		},
+	}
+	listEventsCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", "", "Cluster ID")
+	listEventsCmd.Flags().StringVarP(&ops.since, "since", "", "", "Duration of lookup")
+	listEventsCmd.Flags().BoolVarP(&ops.url, "url", "u", false, "Prints link link url to specific cloudtrail event")
+	listEventsCmd.Flags().BoolVarP(&ops.raw, "raw", "r", false, "Prints Raw cloudtrail event to console")
+	listEventsCmd.Flags().BoolVarP(&ops.all, "all", "A", false, "Prints all cloudtrail events")
+	listEventsCmd.Flags().IntVar(&ops.pages, "pages", 50, "Command will display X pages of Cloud Trail logs for the cluster. Pages is set to 40 by default")
+	listEventsCmd.MarkFlagRequired("cluster-id")
+	return listEventsCmd
+}
+func newWrite_eventsOptions() *LookupEventsoptions {
+
+	return &LookupEventsoptions{}
+
+}
+
+// complete fills in necessary data for the options struct
+func (o *LookupEventsoptions) complete(cmd *cobra.Command, _ []string) error {
+	err := utils.IsValidClusterKey(o.clusterID)
+	if err != nil {
+		return err
+	}
+
+	connection, err := utils.CreateConnection()
+	if err != nil {
+		return err
```

Comment:
**Suggestion:** Here and pretty much every raw `err` return in main logic: wrap the error to make the issue clearer. 

See https://bitfieldconsulting.com/golang/wrapping-errors 
E.g.

```suggestion
	connection, err := utils.CreateConnection()
	if err != nil {
		return fmt.Errorf("unable to create connection to ocm: %w")
```